source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -253,5 +251,37 @@ MStatus MaterialXSurfaceNode::initialize()
 {
     CHECK_MSTATUS(inheritAttributesFrom(MATERIALX_NODE_TYPENAME));
 
+    MFnNumericAttribute nAttr;
+    VP2_TRANSPARENCY_ATTRIBUTE = nAttr.create(
```

Comment:
Turns out Maya expects an input attribute to indicate whether a surface shading DG node is transparent. It can either be a float or float3 and either connected to something upstream or contain a non-0 value.